pptx/opc/package.py

```diff
@@ -104,17 +108,15 @@ def main_document_part(self):
 
     def next_partname(self, tmpl):
         """
-        Return a |PackURI| instance representing the next available partname
+        Return a |PackURI| instance representing the next partname
         matching *tmpl*, which is a printf (%)-style template string
         containing a single replacement item, a '%d' to be used to insert the
         integer portion of the partname. Example: '/ppt/slides/slide%d.xml'
         """
-        partnames = [part.partname for part in self.iter_parts()]
-        for n in range(1, len(partnames) + 2):
-            candidate_partname = tmpl % n
-            if candidate_partname not in partnames:
-                return PackURI(candidate_partname)
-        raise Exception("ProgrammingError: ran out of candidate_partnames")
+        name = tmpl.split(os.sep)[2]
```

Comment:
This seems unreliable. Are you sure you can count on the name always being the third item? Better to figure it is the last item, like `tmpl.split(os.sep)[-1]`. Also, I'm not sure there's a compelling reason to split out the name at all. What is lost by keeping the path in there? The more important thing would be to split before the `%d` so you could match existing part-names with something like `part_name.startswith("/ppt/slides/slide")`, although be careful because `/ppt/slides/slideMaster42.xml` also starts with `"/ppt/slides/slide"`. You'll probably need to use a regular expression or similar method to separate out the numeric suffix.